packages/js-connect-with-portabl/src/index.ts

```diff
@@ -1,21 +1,234 @@
-  root.append(button);
+  async _initiateTransaction(): Promise<void> {
+    const nonce = window.btoa(createRandomString());
+
+    const transactionData = await fetch(
+      `${this.options.connectDomain}/api/v1/agent/${this.options.accountId}/oauth2/transaction`,
+      {
+        method: 'POST',
+        headers: {
+          'Content-Type': 'application/json',
+        },
+        body: JSON.stringify({
+          nonce,
+        }),
+      },
+    );
+
+    const transaction =
+      (await transactionData.json()) as CreateTransactionResponse;
+
+    this.transactionManager.create({ ...transaction, nonce });
+    const transformedAuthRequestUriParams = new URL(transaction.authRequestUri)
+      .search;
+
+    window.location.href = `${this.options.walletDomain}/authorize${transformedAuthRequestUriParams}`;
+  }
+
+  private async _getResponse(
+    responseCode: string,
+  ): Promise<GetResponseDataResponse> {
+    const transaction = this.transactionManager.get();
+    if (!transaction) {
+      throw new Error('No Transaction found');
+    }
+
+    const response = await fetch(
+      `${this.options.connectDomain}/api/v1/agent/${this.options.accountId}/oauth2/response?response_code=${responseCode}&transaction_id=${transaction?.transactionId}`,
+      {
+        method: 'GET',
+      },
+    );
+
+    return response.json();
+  }
+
+  private async _getAccessTokenFromIdToken({
+    idToken,
+  }: {
+    idToken: string;
+  }): Promise<GetAccessTokenResponse> {
+    const response = await fetch(
+      `/${this.options.connectDomain}/api/v1/agent/${this.options.accountId}/oauth2/token`,
+      {
+        method: 'POST',
+        headers: {
+          'Content-Type': 'application/json',
+        },
+        body: JSON.stringify({
+          id_token: idToken,
+          grant_type: 'id_token',
+          scope: 'openid',
+        }),
+      },
+    );
+
+    return response.json();
+  }
+
+  private _validateProof(
+    proof: { challenge: string } | { challenge: string }[],
+  ): boolean {
+    const transaction = this.transactionManager.get();
+    let isValid = true;
+
+    if (Array.isArray(proof)) {
+      for (let i = 0; i < proof.length; i += 1) {
+        const { challenge } = proof[i];
+
+        if (challenge !== transaction?.nonce) {
+          isValid = false;
+          break;
+        }
+      }
+    } else {
+      isValid = proof.challenge === transaction?.nonce;
+    }
+
+    return isValid;
+  }
+
+  private _validateIdToken(idToken: string): boolean {
+    const transaction = this.transactionManager.get();
+    const { nonce } = jwtDecode<{ nonce: string }>(idToken);
+
+    return nonce === transaction?.nonce;
+  }
+
+  async handleRedirectCallback(
+    url: string = window.location.href,
+  ): Promise<void> {
+    const [, queryString] = url.split('?');
+
+    const searchParams = new URLSearchParams(queryString);
+    const responseCode = searchParams.get('response_code');
+    if (!responseCode) {
+      return;
```

Comment:
Add a console warning here to let them know that handleRedirectCallback was invoked but no response_code was found in the url